Model/Payment.php

```diff
@@ -468,13 +468,14 @@ public function refund(InfoInterface $payment, $amount)
                 );
             }
 
-            $refundAmount = CurrencyUtils::toMinor($amount, "USD");
+            $orderCurrency = $order->getOrderCurrencyCode();
+            // $amount argument of refund method is in store currency, we need to get amount from credit memo to get the value in order's currency.
+            $refundAmount = CurrencyUtils::toMinor( $payment->getCreditMemo()->getGrandTotal(), $orderCurrency );
```

Comment:
does the check on L459 still make sense then?

why not remove $amount as an argument then? it sort of seems unnecessary right?

Reply:
I think it still serves as check against invalid call. Also we cannot remove $amount argument because this class is implementing magento's interface so method signature has to be this type

Reply:
ahh thank you for explaining